### ISSUE TITLE
mibRequest.done() async

### ISSUE THREAD

**Issue by affanshahid** [markabrahams/node-net-snmp#90]
It seems I am unable to call `mibRequest.done()` asynchronously when creating an agent. Can provider handlers have async content?

I keep getting the following errror when I call `mibRequest.done` in a `setTimeout()`: 
```
TypeError: Cannot read property 'oid' of undefined
    at writeVarbinds (/app/node_modules/net-snmp/index.js:450:32)
    at GetResponsePdu.SimpleResponsePdu.toBuffer (/app/node_modules/net-snmp/index.js:716:2)
    at Message.toBufferCommunity (/app/node_modules/net-snmp/index.js:1188:11)
    at Message.toBuffer (/app/node_modules/net-snmp/index.js:1173:15)
    at Listener.send (/app/node_modules/net-snmp/index.js:2649:23)
    at Agent.sendResponse (/app/node_modules/net-snmp/index.js:4130:16)
    at MibRequest.mibRequest.done (/app/node_modules/net-snmp/index.js:4023:21)
    at Timeout.setTimeout [as _onTimeout] (/app/src/app2.js:35:11)
    at ontimeout (timers.js:475:11)
    at tryOnTimeout (timers.js:310:5)
```

**Comment by markabrahams** [comment]
Hi @affanshahid - thanks for reporting that.  There was a bug preventing the correct handling mibRequest.done() in an async context.

This is fixed in master now, and the fix is published in version 2.6.6 of the npm.

**Comment by affanshahid** [comment]
Awesome! working now, thanks
